File: Experiment-2/pow.cpp
#include <iostream>
using namespace std;

// Function to calculate num raised to the power of pow using exponentiation by squaring
double power(double num, int pow) {
    // Handle the special case where the power is 0
    if (pow == 0) {
        return 1;  // Any number raised to the power of 0 is 1
    }

    // Handle negative exponents
    if (pow < 0) {
        num = 1.0 / num;
        pow = -pow;
    }

    double result = 1.0;

    while (pow > 0) {
        // If pow is odd, multiply result by num
        if (pow % 2 == 1) {
            result *= num;
        }
        // Square the base
        num *= num;
        // Divide pow by 2
        pow /= 2;
    }

    return result;
}

int main() {
    double num;  // Variable to hold the base number
    cout << "Enter the number: ";
    cin >> num;  // Read the base number from user input

    int pow;  // Variable to hold the exponent value
    cout << "Enter the power of the number: ";
    cin >> pow;  // Read the exponent value from user input

    // Call the power function and store the result
    double result = power(num, pow);

    // Output the result to the user
    cout << "Result: " << result << endl;
    return 0;  // Return 0 to indicate successful execution
}

File: Experiment-2/Pow.py
def power(num, pow):
    # Handle the special case where the power is 0
    if pow == 0:
        return 1  # Any number raised to the power of 0 is 1

    # Handle negative exponents
    if pow < 0:
        # Convert the base number to its reciprocal
        num = 1.0 / num
        # Convert the power to positive
        pow = -pow

    # Initialize result to 1
    result = 1.0

    # Loop to multiply num by itself pow times
    for _ in range(pow):
        result *= num  # Multiply result by num in each iteration

    # Return the final result after pow multiplications
    return result


num = float(input("Enter the number: "))  # Read the base number from user input
pow = int(input("Enter the power of the number: "))  # Read the exponent value from user input

# Call the power function and store the result
result = power(num, pow)

# Output the result to the user
print("Result:", result)